trio/_util.py

```diff
@@ -139,6 +139,7 @@ def _return_value_looks_like_wrong_library(value):
     # function. So we have to just call it and then check whether the
     # return value is a coroutine object.
     # Note: will not be necessary on python>=3.8, see https://bugs.python.org/issue34890
+    # TODO: python3.7 support is now dropped, so the above can be addressed.
```

Comment:
```python
    # We can't check iscoroutinefunction(async_fn), because that will fail
    # for things like functools.partial objects wrapping an async
    # function. So we have to just call it and then check whether the
    # return value is a coroutine object.
    # Note: will not be necessary on python>=3.8, see https://bugs.python.org/issue34890
```

After looking at this for a while, `iscoroutinefunction(async_fn)` breaks 26 tests - and it looks like most, if not all, are due to tests wrapping async methods in sync wrappers - but I think `lowlevel.enable_ki_protection` also breaks it, which seems bad.

Reply:
Opened #2670 - will leave `coroutine_or_error` alone in this PR.